mlrun/model_monitoring/api.py

```diff
@@ -568,9 +576,22 @@ def get_sample_set_statistics(
         # Return the statistics logged with the model:
         return model_artifact_feature_stats
 
-    # Turn the DataItem to DataFrame:
-    if isinstance(sample_set, mlrun.DataItem):
-        sample_set, _ = read_dataset_as_dataframe(dataset=sample_set)
+    # Turn other object types to DataFrame:
+    # A DataFrame is necessary for executing the "drop features" operation.
+    if isinstance(
+        sample_set,
+        (mlrun.DataItem, list, tuple, dict, np.ndarray, pd.Series, pd.DataFrame),
```

Comment:
You can use the `DatasetType` for this, even though it's a `Union`, using `.get_args()`. See https://stackoverflow.com/a/64643971 for a short discussion of this. Seems to be better than copying it again.